### PR TITLE
Week 5 walkthrough files added

### PR DESCRIPTION
**Pull request by thefredo1000**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] PR has a descriptive title
- [ ] PR contains closely related changes only (should not have many unrelated changes)
- [ ] STEP pod members have been added as reviewers
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

### CODE REVIEW

**Comment by seanjohnite** [comment]
Hey Rodrigo, can you send a separate PR to commit all of the file copying, and then your changes in this one?

File: ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      }
+    }
+
+    Collections.sort(conflicts, TimeRange.ORDER_BY_START);
+
+    /**
+     * If the are no events or if there are no conflicts, that means the whole day is free.
+     * We return a time range for the whole day.
+     */
+    if (events.isEmpty() || conflicts.isEmpty()) {
```
Comment (seanjohnite) [inline ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java:51]: looks like if events is empty, then conflicts is also empty? maybe just check conflicts here?

File: ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Iterator;
+
+public final class FindMeetingQuery {
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
```
Comment (seanjohnite) [inline ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java:25]: Slick! I think one way to make this more readable would be to create aptly-named helper functions for each of the steps that are commented here.

File: ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    // If the request is longer than one day we return an empty query.
+    if (request.getDuration() > (24 * 60)) {
+      return returnQuery;
+    }
+
+    /**
+     * Check in every event if the attendees are in the requests.
+     * If true, we add the event to the conflicted set of events.
+     * Then we sert the conflicted time ranges by start.
```
Comment (seanjohnite) [inline ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java:37]: nit: sort

File: ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+     * We return a time range for the whole day.
+     */
+    if (events.isEmpty() || conflicts.isEmpty()) {
+      returnQuery.add(TimeRange.WHOLE_DAY);
+      return returnQuery;
+    } 
+    
+    /**
+     * We go through all of the conflicts and if there is a time difference between the previous
+     * and current conflicts bitter than the duration of the request, we add to the query we are
```
Comment (seanjohnite) [inline ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java:58]: nit: bigger

File: ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+     * goint to return a time range between the ending of the previous event and the begining of
```
Comment (seanjohnite) [inline ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java:59]: You can refer to variable names in comments, "returnQuery" here

File: ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Iterator;
+
+public final class FindMeetingQuery {
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+    Collection<TimeRange> returnQuery = new ArrayList<>();
```
Comment (seanjohnite) [inline ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java:26]: nit: result or resultTimeRanges?

File: ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+      return returnQuery;
+    } 
+    
+    /**
+     * We go through all of the conflicts and if there is a time difference between the previous
+     * and current conflicts bitter than the duration of the request, we add to the query we are
+     * goint to return a time range between the ending of the previous event and the begining of
+     * the new event.
+     */
+    TimeRange previous = TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TimeRange.START_OF_DAY, false);
```
Comment (brettapeters) [inline ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java:62]: You only really need to keep track of the max interval end time seen so far.

```java
int maxEndSoFar = TimeRange.START_OF_DAY;

for (TimeRange current : conflicts) {
  if (current.start() - maxEndSoFar >= request.getDuration()) {
    returnQuery.add(TimeRange.fromStartEnd(maxEndSoFar, current.start(), false));
  }
  maxEndSoFar = Math.max(maxEndSoFar, current.end());
}
```

Comment (thefredo1000) [reply ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java:62]: Thanks for the help, I appreciate it.

File: ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 6 above)
Comment (thefredo1000) [reply ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java:26]: I think resultTimeRanges

File: ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (thefredo1000) [reply ayup/project/src/main/java/com/google/sps/FindMeetingQuery.java:51]: Yes, thank you for noticing

**Pull request merged**